src/test/regress/sql/node_conninfo_reload.sql

```diff
@@ -0,0 +1,154 @@
+-- Make sure changes citus.node_conninfo shutdown connections with old settings
+CREATE SCHEMA node_conninfo_reload;
+SET search_path TO node_conninfo_reload;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+create table test(a int);
+select create_distributed_table('test', 'a');
+
+-- Make sure a connection is opened and cached
+select count(*) from test where a = 0;
+
+show citus.node_conninfo;
+
+-- Set sslmode to something that does not work when connecting
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=doesnotexist';
+select pg_reload_conf();
+select pg_sleep(0.1); -- wait for config reload to apply
+show citus.node_conninfo;
+
+-- Should give a connection error because of bad sslmode
+select count(*) from test where a = 0;
+
+-- Reset it again
+ALTER SYSTEM RESET citus.node_conninfo;
+select pg_reload_conf();
+select pg_sleep(0.1); -- wait for config reload to apply
+show citus.node_conninfo;
+
+-- Should work again
+select count(*) from test where a = 0;
+
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=doesnotexist';
+BEGIN;
+-- Should still work (no SIGHUP yet);
+select count(*) from test where a = 0;
+select pg_reload_conf();
+select pg_sleep(0.1); -- wait for config reload to apply
+show citus.node_conninfo;
+-- Should not fail since a connection was already taken from pool for this
+-- shard, since the same placement is accessed it will reuse that one for this query
+select count(*) from test where a = 0;
+COMMIT;
+
+-- Should fail now with connection error, when transaction is finished
+select count(*) from test where a = 0;
+
+-- Reset it again
+ALTER SYSTEM RESET citus.node_conninfo;
+select pg_reload_conf();
+select pg_sleep(0.1); -- wait for config reload to apply
+show citus.node_conninfo;
+
+-- Should work again
+select count(*) from test where a = 0;
+
+ALTER SYSTEM SET citus.node_conninfo = 'sslmode=doesnotexist';
+BEGIN;
+-- Should still work (no SIGHUP yet);
+INSERT INTO test VALUES(0);
+select pg_reload_conf();
+select pg_sleep(0.1); -- wait for config reload to apply
```

Comment:
Is there a way to force Postgres to wait until all the pending signals are received -- instetad of `pg_sleep`?

Reply:
Tried to find it, but wasn't able to.